modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogParamsValidationUtils.java

```diff
@@ -101,19 +101,28 @@ public static void validateZoneFilter(@Nullable String filter) {
      *
      * @param schema Schema to look up relation with specified name.
      * @param name Name of the relation to look up.
+     * @param checkType Type this check is applicable to.
      * @throws CatalogValidationException If relation with specified name exists in given schema.
      */
-    public static void ensureNoTableIndexOrSysViewExistsWithGivenName(CatalogSchemaDescriptor schema, String name) {
-        if (schema.aliveIndex(name) != null) {
+    public static void ensureNoTableIndexOrSysViewExistsWithGivenName(CatalogSchemaDescriptor schema, String name, int checkType) {
+        if ((checkType & CheckInstanceType.INDEX) != 0 && schema.aliveIndex(name) != null) {
             throw new IndexExistsValidationException(format("Index with name '{}.{}' already exists", schema.name(), name));
         }
 
-        if (schema.table(name) != null) {
+        if ((checkType & CheckInstanceType.TABLE) != 0 && schema.table(name) != null) {
             throw new TableExistsValidationException(format("Table with name '{}.{}' already exists", schema.name(), name));
         }
 
-        if (schema.systemView(name) != null) {
+        if ((checkType & CheckInstanceType.SYS_VIEW) != 0 && schema.systemView(name) != null) {
             throw new CatalogValidationException(format("System view with name '{}.{}' already exists", schema.name(), name));
         }
     }
+
+    /** Types corresponds to appropriate check. */
+    public interface CheckInstanceType {
+        int TABLE = 1;
```

Comment:
How about making `CheckInstanceType` an enum and using a `Set<CheckInstanceType>` (represented with `EnumSet`) instead of a bitmask?